File: src/main/java/com/github/rabend/generators/BooleanGenerator.java
package com.github.rabend.generators;

import com.fasterxml.jackson.databind.JsonNode;

import java.util.concurrent.ThreadLocalRandom;

public class BooleanGenerator extends AbstractValueGenerator {
    @Override
    public String generateRandomValue(final JsonNode node) {
        ThreadLocalRandom random = ThreadLocalRandom.current();

        int randomInt = random.nextInt(0, 2);

        return String.valueOf(randomInt == 0);
    }
}

File: src/main/java/com/github/rabend/generators/BooleanGenerator.kt
package com.github.rabend.generators

import kotlinx.serialization.json.JsonElement
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.JsonPrimitive
import java.util.concurrent.ThreadLocalRandom

class BooleanGenerator : AbstractValueGenerator() {
    override fun generateRandomValue(node: JsonObject): JsonElement {
        val random = ThreadLocalRandom.current()
        val randomInt = random.nextInt(0, 2)
        return JsonPrimitive(randomInt == 0)
    }
}
